lib/config/fileconf.go

```diff
@@ -137,13 +138,28 @@ type SampleFlags struct {
 	ACMEEnabled bool
 	// Version is the Teleport Configuration version.
 	Version string
+	// PublicAddrs sets the hostport the proxy advertises for the HTTP endpoint.
+	PublicAddrs []string
+	// KeyFile is a TLS key file
+	KeyFile string
+	// CertFile is a TLS Certificate file
+	CertFile string
 }
 
 // MakeSampleFileConfig returns a sample config to start
 // a standalone server
 func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
-	if flags.ACMEEnabled && flags.ClusterName == "" {
-		return nil, trace.BadParameter("please provide --cluster-name when using acme, for example --cluster-name=example.com")
+	if (flags.KeyFile == "") != (flags.CertFile == "") { // xor
+		return nil, trace.BadParameter("please provide both --key-file and --cert-file")
+	}
+
+	if flags.ACMEEnabled {
+		if flags.ClusterName == "" {
+			return nil, trace.BadParameter("please provide --cluster-name when using ACME, for example --cluster-name=example.com")
+		}
+		if flags.CertFile != "" {
+			return nil, trace.BadParameter("could not use --key-file/--cert-file when ACME is enabled")
+		}
```

Comment:
nit: It would be nice to introduce a separate function doing only `SampleFlags` flags validation.